x-pack/plugins/observability_solution/apm/public/components/app/mobile/service_overview/geo_map/embedded_map.tsx

```diff
@@ -34,123 +29,19 @@ function EmbeddedMapComponent({
-
-  const {
-    embeddable: embeddablePlugin,
-    maps,
-    notifications,
-  } = useKibana<ApmPluginStartDeps>().services;
-
-  useEffect(() => {
-    async function setupEmbeddable() {
-      const factory = embeddablePlugin?.getEmbeddableFactory<
-        MapEmbeddableInput,
-        MapEmbeddableOutput,
-        MapEmbeddable
-      >(MAP_SAVED_OBJECT_TYPE);
-
-      if (!factory) {
-        setError(true);
-        notifications?.toasts.addDanger({
-          title: i18n.translate('xpack.apm.serviceOverview.embeddedMap.error.toastTitle', {
-            defaultMessage: 'An error occurred when adding map embeddable',
-          }),
-          text: i18n.translate('xpack.apm.serviceOverview.embeddedMap.error.toastDescription', {
-            defaultMessage: `Embeddable factory with id "{embeddableFactoryId}" was not found.`,
-            values: {
-              embeddableFactoryId: MAP_SAVED_OBJECT_TYPE,
-            },
-          }),
-        });
-        return;
-      }
-
-      const input: MapEmbeddableInput = {
-        attributes: { title: '' },
-        id: uuidv4(),
-        title: i18n.translate('xpack.apm.serviceOverview.embeddedMap.input.title', {
-          defaultMessage: 'Latency by country',
-        }),
-        filters,
-        viewMode: ViewMode.VIEW,
-        mapCenter: { lat: 20.43425, lon: 0, zoom: 1.25 },
-        isLayerTOCOpen: false,
-        query: {
-          query: kuery,
-          language: 'kuery',
-        },
-        timeRange: {
-          from: start,
-          to: end,
-        },
-        hideFilterActions: true,
-      };
-
-      const embeddableObject = await factory.create(input);
-
-      setEmbeddable(embeddableObject);
-    }
-
-    setupEmbeddable();
-    // Set up exactly once after the component mounts
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
-
-  // We can only render after embeddable has already initialized
-  useEffect(() => {
-    if (embeddableRoot.current && embeddable) {
-      embeddable.render(embeddableRoot.current);
-    }
-  }, [embeddable, embeddableRoot]);
-
-  useEffect(() => {
-    const setLayerList = async () => {
-      if (embeddable && !isErrorEmbeddable(embeddable) && dataView?.id) {
-        const layerList = await getLayerList({
-          selectedMap,
-          maps,
-          dataViewId: dataView.id,
-        });
-        await Promise.all([embeddable.setLayerList(layerList), embeddable.reload()]);
-      }
-    };
-
-    setLayerList();
-  }, [embeddable, selectedMap, maps, dataView]);
-
-  useEffect(() => {
-    if (embeddable) {
-      embeddable.updateInput({
-        filters,
-        query: {
-          query: kuery,
-          language: 'kuery',
-        },
-        timeRange: {
-          from: start,
-          to: end,
-        },
-      });
-    }
-  }, [start, end, kuery, filters, embeddable, selectedMap]);
+  const layerList = dataView?.id
```

Comment:
You need a `useMemo` here otherwise, layerList will be a new array every time any props change for this Component and that will cause MapComponent to call `setLayerList` when the layerList has not changed